p2p/p2p.proto

```diff
@@ -38,8 +38,17 @@ message AnchorDocumentResponse {
   bool accepted = 1;
 }
 
+enum AccessType {
+  ACCESS_TYPE_INVALID = 0;
+  ACCESS_TYPE_REQUESTER_VERIFICATION = 1;
+  ACCESS_TYPE_NT_OWNER_VERIFICATION = 2;
+  ACCESS_TYPE_ACCESS_TOKEN_VERIFICATION = 3;
+}
+
 message GetDocumentRequest {
   bytes document_identifier = 1;
+  AccessType access_type = 2;
```

Comment:
Quick question, What is the `access_type` used for? Isn't it something we will replace (as you mention) when adding the access token or by default in the read rules of the target document?

Reply:
it's for the handler to know how to validate the GetDocument request.

Reply:
Wouldnt we go through a series of checks instead?

Reply:
This would enable us to have simple switch check and calling specific method of the read access validator